src/comments.js

```diff
@@ -915,7 +919,9 @@ function getCommentChildNodes(node) {
           n !== "precedingNode" &&
           n !== "followingNode"
       )
-      .map((n) => node[n]);
+      .map((n) => node[n])
+      // make sure always AST nodes are returned
+      .filter((n) => typeof n === "object");
```

Comment:
This fixed the `array` tests (we were returning non-AST nodes before)

Reply:
This should also done in Prettier core.